lighthouse-core/config/config.js

```diff
@@ -482,13 +482,21 @@ class Config {
    * @return {!Object} fresh passes object
    */
   static generatePassesNeededByGatherers(oldPasses, requiredGatherers) {
+    const auditsNeedTrace = requiredGatherers.has('traces');
     const passes = JSON.parse(JSON.stringify(oldPasses));
     const filteredPasses = passes.map(pass => {
       // remove any unncessary gatherers from within the passes
       pass.gatherers = pass.gatherers.filter(gathererName => {
         gathererName = GatherRunner.getGathererClass(gathererName).name;
         return requiredGatherers.has(gathererName);
       });
+
+      // disable the trace if no audit requires a trace
+      if (pass.recordTrace && !auditsNeedTrace) {
+        log.warn('config', `Trace not requested by an audit, dropping trace in ${pass.passName}`);
```

Comment:
the only issue with this is that `validatePasses` hasn't run yet, so the pass might not have a `passName` (and we allow no name if there's only one unnamed pass, at which point it should be `DEFAULT_PASS`...ugh)

Maybe '`Trace not requested by an audit, dropping trace in pass '${pass.passName}'`' so at least they get a `pass 'undefined'`? :)

(we should figure out ordering on that at some point, too. Makes sense to validate at the end, so the actual config that's going to be run is sure to be valid, but input is also supposed to be valid so...validate twice?)